gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/rest/HttpJsonServiceStubClassComposer.java

```diff
@@ -1103,6 +1151,124 @@ protected List<Expr> createOperationsStubInitExpr(
+   * @return Expr with the HttpRule built
+   */
+  private Expr createHttpRuleExpr(HttpRule httpRule, boolean checkAdditionalBindings) {
+    Expr httpRuleBuilderExpr =
+        MethodInvocationExpr.builder()
+            .setStaticReferenceType(FIXED_REST_TYPESTORE.get(HttpRule.class.getSimpleName()))
+            .setMethodName("newBuilder")
+            .build();
+
+    httpRuleBuilderExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(httpRuleBuilderExpr)
+            .setMethodName(setMethodFormat(getHttpVerbSetterFromHttpRule(httpRule)))
+            .setArguments(
+                ValueExpr.withValue(StringObjectValue.withValue(getURIValueFromHttpRule(httpRule))))
+            .setReturnType(FIXED_REST_TYPESTORE.get(HttpRule.class.getSimpleName()))
+            .build();
+
+    if (checkAdditionalBindings) {
+      for (HttpRule additionalBindings : httpRule.getAdditionalBindingsList()) {
+        httpRuleBuilderExpr =
+            MethodInvocationExpr.builder()
+                .setExprReferenceExpr(httpRuleBuilderExpr)
+                .setMethodName("addAdditionalBindings")
+                .setArguments(Arrays.asList(createHttpRuleExpr(additionalBindings, false)))
+                .build();
+      }
+    }
+
+    httpRuleBuilderExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(httpRuleBuilderExpr)
+            .setMethodName("build")
+            .setReturnType(FIXED_REST_TYPESTORE.get(HttpRule.class.getSimpleName()))
+            .build();
+    return httpRuleBuilderExpr;
+  }
+
+  /**
+   * Parses the Service Yaml file's for custom HttpRules. Filter the rules based on a predicate to
+   * match for certain custom rules.
+   *
+   * @param context GapicContext that contains the Service Yaml information
+   * @param predicate HttpRule predicate to filter for certain rules
+   * @return Map contains KV pairing of (HttpRule Selector -> HttpRule object)
+   */
+  private Map<String, HttpRule> parseCustomHttpRules(
+      GapicContext context, Predicate<HttpRule> predicate) {
+    com.google.api.Service service = context.serviceYamlProto();
+    if (service == null || service.getHttp() == null) {
+      return ImmutableMap.of();
+    }
+    return service.getHttp().getRulesList().stream()
+        .filter(predicate)
+        .collect(Collectors.toMap(HttpRule::getSelector, x -> x));
+  }
+
+  /**
+   * Parse the HttpRule for the HttpVerb containing the custom URI. Concat the Http Verb to return
+   * the setting function name.
+   *
+   * <p>This ignores the `Custom` HttpRule. Operation proto only supports GET, POST, DELETE and
+   * Custom requires special logic.
+   *
+   * @param httpRule HttpRule to parse
+   * @return HttpRule.Builder's setter function name when setting for an HttpVerb
+   */
+  private String getHttpVerbSetterFromHttpRule(HttpRule httpRule) {
+    switch (httpRule.getPatternCase().getNumber()) {
```

Comment:
Can we use `httpRule.getPatternCase().toString()` for this case?

Reply:
Ah yeah, I think so. Let me try changing to that.